### DISCUSSION TITLE
MinIO Gateway deprecation

### DISCUSSION THREAD

**Discussion by JonMerlevede** [juicedata/juicefs#2786] [General]
[JuiceFS S3 Gateway](https://juicefs.com/docs/community/s3_gateway/) is based on MinIO Gateway, which is, unfortunately [deprecated](https://blog.min.io/deprecation-of-the-minio-gateway/). No new MinIO Gateway versions will be released. Forking is encouraged but the GNU AGPL v3 license is required.

**What would you like to be added**: A roadmap for JuiceFS S3 Gateway.

**Why is this needed**: Deprecation of the component MinIO Gateway.

**Comment by davies** [comment]
@JonMerlevede We are using a forked MinIO [1] before license change, and we will continue to maintain this fork for security and bug fix.

[1] https://github.com/juicedata/minio

**Reply by mon-jai** [reply]
Seems like it is AGPL?

**Reply by davies** [reply]
The full featured S3 gateway is under AGPL, but the simple one embedded in JuiceFS is under Apache License 2.

**Reply by davies** [reply]
see https://github.com/juicedata/minio/blob/8866d5c9df8cdb1027844b0eb4e2d84d3c144008/LICENSE

**Reply by mon-jai** [reply]
Oh I see. It is a little bit hard to find because `master` isn't the default branch for the repo.

**Comment by JonMerlevede** [comment]
That's great news, thanks for clearing that up for me!
